### ISSUE TITLE
Performance Matrix

### ISSUE THREAD

**Issue by bmaschio** [jolie/website#10]
@jolie/developers 
Once again from a "Marketing" point of view Jolie is a bit of unknown quantity.
The good point that a first glance it looks clean and tidy in its syntax and structure ( thanks to the good formal approach ), yet there is no data on how it performs on some keys technical indicator.
Such as
- Memory occupation ( static and dynamic) 
- Response Times 
- Performance on stress test  

Now I do realize that probably there is plenty of data on these issues, but it would be nice to filter some of them to the general public
I could help to collect some data on our site where jolie is used for several application

Regards

Balint

**Comment by fmontesi** [comment]
I fully agree. Some performance graphs are coming soon (I have a comparison
with apache+php and JavaEE), and I know how much memory is consumed by
Jolie, but it would be nice to make a page that is easy to understand.

I would start with memory and throughput. We could test memory per service
and per open session, for example.

However, we should be careful. People are very suspicious of benchmarking
results (rightfully so!) so we should probably publish our benchmark code
and how we measure stuff. I'd take the occasion to build a benchmarking
suite to help us track how Jolie's memory consumption and performance vary
with each release. I do my own internal tests at each release, but it would
be nice to have something more standard. What do you think? I could open a
repo for this where we all collaborate. I think it's an interesting problem
for many, so we could accept also new benchmarks whenever somebody wants to
test something in particular.

**Comment by bmaschio** [comment]
@fmontesi 
I think the key word there is transparency ( a bit like in academical papers ;) ) 
Yes I think a new repository could be a good way to go. 
Defining some standard testing procedure for different application area such as : 
- HTTP Server
- Database access server
- Application Server ( where the "business" logic lives )
- Pseudo ESB application

I think it would be nice also to distill a best practice in terms of architecture ecc ecc

**Comment by mwallnoefer** [comment]
@fmontesi ping, has anything been done in the meantime? The `Roadmap` link does not work any more.
